src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -164,8 +164,14 @@ public function createCheckoutsPaginator(array $criteria = null, array $sorting
      */
     public function countByCustomer(CustomerInterface $customer)
     {
-        $count = $this->createByCustomerQueryBuilder($customer)
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        $count = $queryBuilder
             ->select('count(o.id)')
+            ->andWhere('o.customer = :customer')
+            ->andWhere($queryBuilder->expr()->notIn('o.state', ':states'))
+            ->setParameter('customer', $customer)
+            ->setParameter('states', [OrderInterface::STATE_CART, OrderInterface::STATE_CANCELLED])
```

Comment:
Not sure if we should define it here. Method `countByCustomer` suggest that it will count all customer method. We should change a name of this method. And consider an additional parameter